rustls/src/verify.rs

```diff
@@ -637,18 +637,7 @@ pub(crate) fn pki_error(error: webpki::Error) -> Error {
 }
 
 /// Turns off client authentication.
-pub struct NoClientAuth;
-
-impl NoClientAuth {
-    /// Construct a [`NoClientAuth`], wrap it in an [`Arc`] and coerce it to
-    /// `dyn ClientCertVerifier`.
-    #[inline(always)]
-    pub fn boxed() -> Arc<dyn ClientCertVerifier> {
-        // This function is needed because `ClientCertVerifier` is only reachable if the
-        // `dangerous_configuration` feature is enabled, which makes coercing hard to outside users
-        Arc::new(Self)
-    }
-}
+pub(crate) struct NoClientAuth;
```

Comment:
So the reason we're keeping this around is a little subtle when reading the source, so maybe there should be some comments (here but maybe also around the builder and/or the `WebpkiClientVerifier` type?): with `WebPkiClientVerifier` in `anon_policy: Allow` mode the server still **offers** client authentication, which is not the case for `NoClientAuth`. I think we should probably also call out this difference in the docstring for the builder's `build()` method.

Reply:
I've updated some comments to emphasize this aspect. Good call out :+1: